Voycar.Api.Web.Tests.Integration/Reservations/Get_Personal/Tests.cs

```diff
@@ -0,0 +1,255 @@
+
+        // Cleanup
+        this.Context.RemoveRange(reservations);
+        await this.Context.SaveChangesAsync();
+    }
+
+
+    [Fact]
+    public async Task Get_Request_ReturnsOk_And_EmptyList()
+    {
+        // Act
+        var (httpResponse, response) =
+            await this._app.Member.GETAsync<P.Endpoint, P.Request, P.Response>(new P.Request());
+
+        // Assert
+        httpResponse.StatusCode.Should().Be(HttpStatusCode.OK);
+        response.expired.Count.Should().Be(0);
+        response.active.Count.Should().Be(0);
+        response.planned.Count.Should().Be(0);
+    }
+
+
+    [Fact]
+    public async Task Get_Personal_AsEmployee_Returns_Forbidden()
+    {
+        // Act
+        var (httpResponse, _) = await this._app.Employee.GETAsync<P.Endpoint, P.Request, P.Response>(new P.Request());
+
+        // Assert
+        httpResponse.StatusCode.Should().Be(HttpStatusCode.Forbidden);
+    }
+
+
+    [Fact]
+    public async Task Get_Personal_AsAdmin_Returns_Forbidden()
+    {
+        // Act
+        var (httpResponse, _) = await this._app.Admin.GETAsync<P.Endpoint, P.Request, P.Response>(new P.Request());
+
+        // Assert
+        httpResponse.StatusCode.Should().Be(HttpStatusCode.Forbidden);
+    }
+
+
+    private void ValidateExpiredReservations(P.Response response, List<Reservation> reservations)
+    {
+        foreach (var reservation in reservations)
+        {
+            var responseReservation = response.expired.FirstOrDefault(r => r.Id == reservation.Id);
+            responseReservation.Should().NotBeNull();
+
+            responseReservation!.Begin.ToShortTimeString().Should().Be(reservation.Begin.ToShortTimeString());
+            responseReservation.End.ToShortTimeString().Should().Be(reservation.End.ToShortTimeString());
+        }
+    }
+
+
+    private void ValidateActiveReservations(P.Response response, List<Reservation> reservations)
+    {
+        foreach (var reservation in reservations)
+        {
+            var responseReservation = response.active.FirstOrDefault(r => r.Id == reservation.Id);
+            responseReservation.Should().NotBeNull();
+
+            responseReservation!.Begin.ToShortTimeString().Should().Be(reservation.Begin.ToShortTimeString());
+            responseReservation.End.ToShortTimeString().Should().Be(reservation.End.ToShortTimeString());
+        }
+    }
+
+
+    private void ValidatePlannedReservations(P.Response response, List<Reservation> reservations)
+    {
+        foreach (var reservation in reservations)
+        {
+            var responseReservation = response.planned.FirstOrDefault(r => r.Id == reservation.Id);
+            responseReservation.Should().NotBeNull();
+
+            responseReservation!.Begin.ToShortTimeString().Should().Be(reservation.Begin.ToShortTimeString());
+            responseReservation.End.ToShortTimeString().Should().Be(reservation.End.ToShortTimeString());
+        }
+    }
```

Comment:
Diese drei Methoden bitte `static` machen und an den Beginn der Klasse verschieben (hinter den Konstruktor, wo auch die andere Hilfsmethode ist).

Wenn ohne zu großen Aufwand möglich auch noch redundanten Code der drei Methoden vermeiden.

Und warum `FirstOrDefault` anstatt nur `First`? Ergibt es Sinn, dass du nen Default Value weiterverarbeitest, wenn nichts in der Response gefunden wurde?